### ISSUE TITLE
[terra-time-input] Add TimeInputField subcomponent

### ISSUE THREAD

**Issue by neilpfeiffer** [cerner/terra-framework#1246]
# Feature Request

## Description
Create a new "_Field_" composition subcomponent to make a`TimeInputField` in 'terra-time-input'.
A new fieldset will need to be created with a legend, and matching help and error text props similar to the terra-field component and other pseudo _field_ compositions, similar to DatePickerField and DateInputField.

The Time format mask can then be re-applied appropriately in the `help` text prop, resolving the alignment/layout issues caused if a consumer makes their own time field on their own, and provide for improved accessibility. 

## Additional Context / Screenshots

| current: consumer created layout | same with field set to isInvalid / error |
| ---- | ---- |
| ![Screen Shot 2020-09-11 at 9 48 50 PM](https://user-images.githubusercontent.com/1355485/93063241-94501100-f63b-11ea-8305-a156ff9ee325.png) | ![Screen Shot 2020-09-11 at 9 44 50 PM](https://user-images.githubusercontent.com/1355485/93063285-a467f080-f63b-11ea-8266-1660a3e7f244.png) |

| desired: layout for TimeInputField | desired with field set to isInvalid / error |
| ---- | ---- |
| ![Screen Shot 2020-09-11 at 9 48 50 PM](https://user-images.githubusercontent.com/1355485/93063357-ba75b100-f63b-11ea-9759-780c8bd2b6b2.png) | ![Screen Shot 2020-09-11 at 9 46 59 PM](https://user-images.githubusercontent.com/1355485/93063378-c19cbf00-f63b-11ea-8a10-86876573d609.png) |

**Comment by pranav300** [comment]
## React Props For Time Input Field:

| Prop | Type | Default | Description |
|----------|------|---------|--------------|
dateInputId | string | required | none | The timeInput identifier. Links the htmlFor of the field to the select identifier.
disabled | bool | optional | false | Whether the time input should be disabled.
error | node | optional | null | Error message for when the input is invalid. This will only be displayed if isInvalid is true.
errorIcon | element | optional | <IconError /> | Error Icon to display when the input is invalid.
help | node | optional | null | Help element to display with the input.
hideRequired | bool | optional | false | Whether or not to hide the required indicator on the label.
hourAttributes | object | optional | {} | Custom input attributes to apply to the hour input
htmlFor | string | optional | none | The htmlFor attribute on the field label.
inputAttributes | object | optional | undefined | Custom input attributes that apply to the hour, minute, and second inputs.
isIncomplete | bool | optional | false | Whether the input displays as Incomplete. Use when no value has been provided. (usage note: required must also be set).
isInline | bool | optional | false | Whether or not the field is an inline field.
isInvalid | bool | optional | false | Whether the input displays as Invalid. Use when value does not meet validation pattern.
isInvalidMeridiem | bool | optional | false | Whether the selected meridiem displays as Invalid. Use when value does not meet validation pattern.
isLabelHidden | bool | optional | false | Whether or not the label is visible. Use this props to hide a label while still creating it on the DOM for accessibility.
label | string | required | none | The label of the form control children.
labelAttrs | object | optional | {} | Attributes to attach to the label.
maxWidth | string | optional | undefined | Set the max-width of a field using length or %. Best practice recommendation to never exceed a rendered value of 1020px. (Note: Providing custom inline styles will take precedence.)
minuteAttributes | object | optional | {} | Custom input attributes to apply to the minutes input
name | string | required | none | Name of the time input. The name should be unique.
onBlur | func | optional | null | A callback function to execute when the entire time input component loses focus. This event does not get triggered when the focus is moved from the hour input to the minute input or meridiem because the focus is still within the main time input component.
onChange | func | optional | null | A callback function to execute when either the hour or minute has been changed. The first parameter is the event. The second parameter is the changed time value.
onFocus | func | optional | undefined | A callback function triggered when the hour input or minute input receives focus.
refCallback | func | optional | undefined | Callback ref to pass into the input dom element.
required | bool | optional | false | Whether or not the time is required.
secondAttributes | object | optional | {} | Custom input attributes to apply to the seconds input
showOptional | bool | optional | false | Whether or not to append the 'optional' label to a non-required field label.
showSeconds | bool | optional | false | Whether the input for seconds should be displayed or not. If true then the second field must have a valid number for the overall time to be considered valid.
value | string | optional | undefined | An ISO 8601 string representation of the time value in the input.
variant | enum <br />`[ "TimeUtil.FORMAT_12_HOUR", "TimeUtil.FORMAT_24_HOUR"]`| optional | TimeUtil.FORMAT_24_HOUR | Type of time input to initialize. Must be 24-hour or 12-hour. The de, es-ES, es, fr-FR, fr, nl-BE, nl, pt-BR, pt, sv-SE and sv locales do not use the 12-hour time notation. If the variant prop if set to 12-hour for one of these supported locales, the variant will be ignored and defaults to 24-hour.

## Example:
```jsx
import React from 'react';
import classNames from 'classnames/bind';
import TimeInputField from '../../../../TimeInputField';
import styles from '../common/TimeInput.test.module.scss';

const cx = classNames.bind(styles);

const TimeInputFieldDefault = () => (
  <div className={cx('content-wrapper')}>
    <TimeInputField
      dateInputId="timeInput"
    />
  </div>
);

export default TimeInputFieldDefault;
```

**Comment by pranav300** [comment]
Hello Everyone,
So, I was planning to remove prop `inputAttributes` as we have already provided props for individual inputs `hourAttributes, minuteAttributes, secondAttributes`.

If we deprecate the prop then it will cause unnecessary complications for the conditions of description ids.

And this prop seems redundant.

Any thoughts on whether we should remove it or deprecate it?

**Comment by ShettyAkarsh** [comment]
We should update 'dateInputId' prop to 'timeInputId'. also regarding the prop errorIcon are we using this prop here ? like in datepickerfield also we have this prop but i am not seeing that being used anywhere.

**Comment by pranav300** [comment]
> also regarding the prop errorIcon are we using this prop here ? like in datepickerfield also we have this prop but i am not seeing that being used anywhere.

@ShettyAkarsh, Yup this prop is not being used anywhere in time-input and date-picker. I took it from the date-picker and was waiting to ask this question in the draft PR.

**Comment by benbcai** [comment]
Would you need the private `intl` prop to construct the help label similar to [this](https://github.com/cerner/terra-framework/blob/main/packages/terra-date-picker/src/DatePickerField.jsx#L239-L249)?

**Comment by benbcai** [comment]
👍

**Comment by pranav300** [comment]
> Are legend, legendAttributes, and isLegendHidden needed for this given that it is a group of inputs but appears as one?

We are adding form-field to timeinputfield. Therefore the props label, labelAttributes, and isLabelHidden do the same task.
